ksql-engine/src/main/java/io/confluent/ksql/function/udaf/max/IntegerMaxKudaf.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.function.udaf.max;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.streams.kstream.Merger;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import io.confluent.ksql.function.KsqlAggregateFunction;
+import io.confluent.ksql.parser.tree.Expression;
+
+public class IntegerMaxKudaf extends KsqlAggregateFunction<Integer, Integer> {
+
+  public IntegerMaxKudaf(Integer argIndexInValue) {
+    super(argIndexInValue, () -> Integer.MIN_VALUE, Schema.INT32_SCHEMA,
+          Arrays.asList(Schema.INT32_SCHEMA)
+    );
+  }
+
+  @Override
+  public Integer aggregate(Integer currentVal, Integer currentAggVal) {
```

Comment:
Are `nulls` supported in any way, or is a NPE what's expected? Do we want a more descriptive error than just NPE?

Reply:
Yes, for this implementation we just ignore null values as Postgres does. But we will need to decide on how to handle null values consistently in KSQL. We track this here: https://github.com/confluentinc/ksql/issues/750